Sources/Purchasing/StoreKitAbstractions/Test Data/TestStoreProduct.swift

```diff
@@ -0,0 +1,147 @@
+///     localizedTitle: "PRO monthly",
+///     price: 3.99,
+///     localizedPriceString: "$3.99",
+///     productIdentifier: "com.revenuecat.product",
+///     productType: .autoRenewableSubscription,
+///     localizedDescription: "Description",
+///     subscriptionGroupIdentifier: "group",
+///     subscriptionPeriod: .init(value: 1, unit: .month)
+/// )
+///
+/// let offering = Offering(
+///     identifier: "offering",
+///     serverDescription: "Main offering",
+///     metadata: [:],
+///     availablePackages: [
+///         .init(
+///             identifier: "monthly",
+///             packageType: .monthly,
+///             storeProduct: product.toStoreProduct(),
+///             offeringIdentifier: offering
+///         ),
+///     ]
+/// )
+/// ```
+public struct TestStoreProduct {
+
+    // Note: this class inherits its docs from `StoreProductType`
+    // swiftlint:disable missing_docs
+
+    public var localizedTitle: String
+    public var price: Decimal
+    public var localizedPriceString: String
+    public var productIdentifier: String
+    public var productType: StoreProduct.ProductType
+    public var localizedDescription: String
+    public var subscriptionGroupIdentifier: String?
+    public var subscriptionPeriod: SubscriptionPeriod?
+    public var introductoryDiscount: StoreProductDiscount?
+    public var discounts: [StoreProductDiscount]
+
+    public init(
+        localizedTitle: String,
+        price: Decimal,
+        localizedPriceString: String,
+        productIdentifier: String,
+        productType: StoreProduct.ProductType,
+        localizedDescription: String,
+        subscriptionGroupIdentifier: String? = nil,
+        subscriptionPeriod: SubscriptionPeriod? = nil,
+        introductoryDiscount: TestStoreProductDiscount? = nil,
+        discounts: [TestStoreProductDiscount] = []
+    ) {
+        self.localizedTitle = localizedTitle
+        self.price = price
+        self.localizedPriceString = localizedPriceString
+        self.productIdentifier = productIdentifier
+        self.productType = productType
+        self.localizedDescription = localizedDescription
+        self.subscriptionGroupIdentifier = subscriptionGroupIdentifier
+        self.subscriptionPeriod = subscriptionPeriod
+        self.introductoryDiscount = introductoryDiscount?.toStoreProductDiscount()
+        self.discounts = discounts.map { $0.toStoreProductDiscount() }
+    }
+
+    // swiftlint:enable missing_docs
+
+    private let priceFormatterProvider: PriceFormatterProvider = .init()
+
+}
+
+// Ensure consistency with the internal type
+extension TestStoreProduct: StoreProductType {
+
+    internal var productCategory: StoreProduct.ProductCategory { return self.productType.productCategory }
+
+    internal var currencyCode: String? {
+        // Test currency defaults to current locale
+        return Locale.current.rc_currencyCode
+    }
+
+    internal var isFamilyShareable: Bool { return false }
```

Comment:
maybe later we could expose this one as a stored property, so that you can also check flows where you have a paywall that states whether the subs is family shareable or not

Reply:
Oh let me add it. I removed this parameter cause I didn't think it was useful to mock, but you make a good point.